drivers/sensor/adt7420/Kconfig

```diff
@@ -0,0 +1,117 @@
+# Kconfig - ADT7420 temperature sensor configuration options
+#
+# Copyright (c) 2018 Analog Devices Inc.
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+menuconfig ADT7420
+	bool
+	prompt "ADT7420 Temperature Sensor"
+	depends on I2C
+	help
+	  Enable the driver for Analog Devices ADT7420 High-Accuracy
+	  16-bit Digital I2C Temperature Sensors.
+
+config ADT7420_NAME
```

Comment:
Hi Maureen,

Thanks for your feedback!

In theory device tree is used to describe bus hierarchies and devices that can appear under a bus. 
Basically it describes the physical devices in a system, which cannot be automatically probed.

I totally agree that the parent bus and slave addresses, etc. should come from the device tree.
Practically this would come from the parent device which will instantiate all its devices, via a device specific structure.
In fact that’s the only way to allow instantiating the same driver multiple times…
Which I have no idea, how that can be done in Zephyr today.
 
This being said,  Zephyr device tree usage is very limited. The user must provide a fixup file which maps
device tree properties to CONFIG options.

So my question is what do you mean by “should come from the device tree” ?
Isn’t what’s implemented in this driver exactly the same what all other drivers/sensor do?

Reply:
It's been a long process converting drivers over to device tree, so you will find that some drivers are further along than others. For new drivers, however, we do want to enforce getting hardware information from device tree from the start. Please have a look at `drivers/sensors/fxos8700` and `boards/arm/frdm_k64f/frdm_k64f.dts` for an example of what we're looking for.

Reply:
Oh I see - you asking me to guard these config options with "if !HAS_DTS_I2C_DEVICE" !?

Reply:
Yes, and to add a binding in `dts/bindings/sensor/`